hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/event/SMFilterEvent.java

```diff
@@ -8,19 +8,60 @@
  */
 package org.eclipse.hawkbit.ui.artifacts.event;
 
+import org.eclipse.hawkbit.ui.common.ManagementEntityState;
+
 /**
- * Software module filter events.
- *
- *
- *
+ * Software module filter event. Is thrown when there is a filter action on a
+ * software module table on the Distribution or Upload View. It is possible to
+ * filter by text or type.
  */
-public enum SMFilterEvent {
+public class SMFilterEvent {
+
+    /**
+     * Enum for the different types of SMFilterEvent
+     */
+    public enum SMFilterEventType {
+        /**
+         * filters the software modules by type
+         */
+        REMOVE_FILTER_BY_TYPE,
+        /**
+         * remove software modules' filter by type
+         */
+        FILTER_BY_TYPE,
+        /**
+         * filters the software modules by text
+         */
+        FILTER_BY_TEXT,
+        /**
+         * remove software modules' filter by text
+         */
+        REMOVE_FILTER_BY_TEXT;
+    }
+
+    private final SMFilterEventType eventType;
+
+    private final ManagementEntityState<Long> origin;
```

Comment:
I'm not sure, if it is a good idea to send the whole managementEntityState around.